lib/random_tools.rb

```diff
@@ -0,0 +1,40 @@
+module Upaya
+  module RandomTools
+    # rubocop:disable MethodLength, Style/IfUnlessModifier
+
+    # Randomly choose an item among choices with integer weights. For example,
+    # if passed `{'a' => 3, 'b' => 1}`, the expected return value will approach
+    # 'a' 75% of the time and 'b' 25% of the time.
+    #
+    # @param [Hash{Object => Integer}] choices A mapping from each choice to an
+    #   integer weight. The weights will be relative to the sum of all weights.
+    #
+    # @return [Object] One of the keys chosen from the choices hash.
+    #
+    def self.random_weighted_sample(choices)
+      if choices.empty?
+        raise ArgumentError, 'Cannot choose among empty choices hash'
+      end
+
+      sum = 0
+      choices.each_pair do |item, weight|
+        unless weight.is_a?(Integer)
```

Comment:
Rather than performing validation in here, what do you think about doing it in `lib/config_validator`?